Documentation/spec/docfx_document_schema.md

```diff
@@ -1,126 +1,161 @@
 
-If a processor need to do some operations on a specified property, while the location of the property differs with schemas, it's hard to share the code.
-In this case, a tag can be defined to mark the property, then the processor can fetch the instances with defined tag from schema parser, and read or modify it.
+## 6. Keywords in detail
 
-### 6.3 d-overwrite
-The value of "overwrite" MUST be a string. It specifies how DocFX uses the [overwrite files](..\tutorial\intro_overwrite_files.md) to overwrite the existing values.
+### 6.1 type
 
-The following values are allowed:
-* `replace`: The value will be overwitten by the one specified in overwrite files.
-* `key`: If the instance is a string, it will be used as the key of the current object. It can be used to identify the object to overwrite in an array.
-* `merge`: If the instance is an object, it will be merged with the one in overwrite files. If there are duplicated keys, the one in overwrite files will overwrite the existing one.
-* `ignore`: Mark the current instance cannot be overwritten.
+Same as in JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.25
 
-For `object` and `array` type, the default behavior is `merge`. For other types, the default behavior is `replace`.
+> The value of this keyword MUST be either a string or an array. If it is an array, elements of the array MUST be strings and MUST be unique.
+>
+> String values MUST be one of the six primitive types ("null", "boolean", "object", "array", "number", or "string"), or "integer" which matches any number with a zero fractional part.
+>
+> An instance validates if and only if the instance is in any of the sets listed for this keyword.
+
+### 6.2 d-include
+It defines whether current property is a file path, whose content SHOULD be included as the value of the property.
+
+### 6.3 d-contentType
+It defines how applications interpret the property. The values MUST be one of the following:
 
-> [!Note]
-> After applying the overwrite files, the model should be checked again to ensure that all the validation rules still satisfy.
+| Value      | Description
+|------------|-------------
+| `uid`      | `type` MUST be `string`. It means the property defines a unique identifier inside current document model
+| `fileLink` | `type` MUST be `string`. It means the property defines a file link inside current document model. Application CAN help to validate if the linked file exists, and update the file link if the linked file changes its output path.
+| `uidLink`  | `type` MUST be `string`. It means the property defines a UID link inside current document model. Application CAN help to validate if the linked UID exists, and resolve the UID link to the corresponding file output path.
+| `markdown` | `type` MUST be `string`. It means the property is in [DocFX flavored Markdown](..\spec\docfx_flavored_markdown.md) syntax. Application CAN help to transform it into HTML format.
```

Comment:
How can we describe `includeFile` and `includeMarkdownFile` in this new proposal?